File: web/src/api/MyNFT.js
import http from '../utils/request'

// 铸造NFT
export const mintNFT = (data) => http.post('/mynft/mint', data)
// 批量铸造
export const bulkMintNFT = (data) => http.post('/mynft/bulk-mint', data)
// 设置合约元数据
export const setContractMetadata = (data) => http.post('/mynft/set-metadata', data)
// 更新版税
export const updateRoyalty = (data) => http.post('/mynft/update-royalty', data)
// 查询tokenURI
export const getTokenURI = (tokenId) => http.get(`/mynft/token-uri/${tokenId}`)
// 查询合约支持的接口
export const supportsInterface = (interfaceId) => http.get(`/mynft/supports-interface/${interfaceId}`)
// 查询owner
export const getOwner = () => http.get('/mynft/owner')
// 查询余额
export const getBalanceOf = (address) => http.get(`/mynft/balance-of/${address}`)
// 查询ownerOf
export const getOwnerOf = (tokenId) => http.get(`/mynft/owner-of/${tokenId}`)
// 查询总供应量
export const getTotalSupply = () => http.get('/mynft/total-supply') 

File: server/router/mynft.go
package router

import (
	"server/api"

	"github.com/gin-gonic/gin"
)

func InitMyNFTRouter(Router *gin.RouterGroup) {
	mynftApi := api.MyNFTApi{}

	// MyNFT 铸造相关接口
	Router.POST("/mynft/mint", mynftApi.MintNFT)                           // 铸造单个NFT
	Router.POST("/mynft/bulk-mint", mynftApi.BulkMintNFT)                  // 批量铸造NFT
	Router.POST("/mynft/set-metadata", mynftApi.SetContractMetadata)       // 设置合约元数据
	Router.POST("/mynft/update-royalty", mynftApi.UpdateRoyaltyPercentage) // 更新版税百分比

	// MyNFT 查询相关接口
	Router.GET("/mynft/token-uri", mynftApi.GetTokenURI)                // 查询token URI
	Router.GET("/mynft/supports-interface", mynftApi.SupportsInterface) // 查询接口支持
	Router.GET("/mynft/owner", mynftApi.GetOwner)                       // 查询合约owner
	Router.GET("/mynft/balance-of", mynftApi.GetBalanceOf)              // 查询地址余额
	Router.GET("/mynft/owner-of", mynftApi.GetOwnerOf)                  // 查询token所有者
	Router.GET("/mynft/total-supply", mynftApi.GetTotalSupply)          // 查询总供应量
}
